ggu/main.c

```diff
@@ -16,11 +16,380 @@
  * along with this program.  If not, see <http://www.gnu.org/licenses/>.
  */
 
+#include <stdbool.h>
+
+#include <rte_ip.h>
+#include <rte_udp.h>
+#include <rte_log.h>
+#include <rte_lcore.h>
+#include <rte_malloc.h>
+#include <rte_ethdev.h>
+#include <rte_atomic.h>
+
+#include "gatekeeper_gk.h"
 #include "gatekeeper_ggu.h"
+#include "gatekeeper_net.h"
+#include "gatekeeper_main.h"
+#include "gatekeeper_config.h"
+
+#define GGU_PD_VER1 (1)
+
+static rte_atomic16_t num_ggu_conf_alloc = RTE_ATOMIC16_INIT(0);
```

Comment:
The variable `num_ggu_conf_alloc` is only used in `alloc_ggu_conf()`. Given that the variable is static, you can move it to the body of the function.